modules/vault/vault.go

```diff
@@ -0,0 +1,104 @@
+package vault
+
+import (
+	"context"
+	"fmt"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"strings"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/testcontainers/testcontainers-go"
+)
+
+const (
+	defaultPort      = "8200"
+	defaultImageName = "vault:1.13.0"
```

Comment:
Because of the notice below (source: https://hub.docker.com/_/vault) we should point to `hashicorp/vault` instead.

> DEPRECATION NOTICE
Upcoming in Vault 1.14, we will stop publishing official Dockerhub images and publish only our Verified Publisher images. Users of Docker images should pull from [hashicorp/vault](https://hub.docker.com/r/hashicorp/vault) instead of [vault](https://hub.docker.com/_/vault). Verified Publisher images can be found at https://hub.docker.com/r/hashicorp/vault.

```suggestion
	defaultImageName = "hashicorp/vault:1.13.0"
```